oauth2_provider/oauth2_validators.py

```diff
@@ -237,6 +239,57 @@ def validate_client_id(self, client_id, request, *args, **kwargs):
     def get_default_redirect_uri(self, client_id, request, *args, **kwargs):
         return request.client.default_redirect_uri
 
+    def _get_token_from_authentication_server(self, token):
+        try:
+            bearer = 'Bearer {}'.format(oauth2_settings.RESOURCE_SERVER_AUTH_TOKEN)
+            resp = requests.post(url=oauth2_settings.RESOURCE_SERVER_INTROSPECTION_URL,
+                                 data={'token': token},
+                                 headers={'Authorization': bearer})
+            content = resp.json()
+        except requests.exceptions.RequestException as e:
+            log.error("Introspection: failed to lookup token: %s" % e.message)
+        except ValueError as e:
+            log.error("Introspection: failed to parse response: %s" % e.message)
```

Comment:
Use `log.exception()` without e.message, that will let you keep the whole traceback.